### PR TITLE
Move build stream to use polymer-build

### PR DESCRIPTION
**Pull request by FredKSchott**
~~*Note: Currently using a github tag instead of npm in package.json for polymer-build. Will be removed once package is updated on npm*~~

This moves our current build logic over to the polymer-build pipeline. In addition to the much more efficient build pipeline, polymer-build should introduce several bug fixes and bring us back to maintaining a single build system.

/cc @justinfagnani

### CODE REVIEW

**Comment by FredKSchott** [comment]
@justinfagnani PTAL, actively using polymer-build in the CLI should be useful for understanding its usage

File: src/build/build.ts [hunk 1]
```diff
@@ -16,86 +16,48 @@ import * as gutil from 'gulp-util';
...
-    foundGulpfileDir = findup.sync(process.cwd(), 'gulpfile.js');
-  } catch (e) {
-    logger.debug(`no gulpfile found (searched up from ${process.cwd()})`);
-    return null;
-  }
-
-  let gulpfulePath = path.join(foundGulpfileDir, 'gulpfile.js');
-  logger.debug('found gulpfile', { path: gulpfulePath });
-  return require(gulpfulePath);
+export interface DepsIndex {
```
Comment (justinfagnani) [inline src/build/build.ts]: Doesn't this interface exist in polymer-build?

File: src/build/build.ts [hunk 2]
```diff
@@ -16,86 +16,48 @@ import * as gutil from 'gulp-util';
...
-      '!bower_components/**/*',
-      '!node_modules',
-      '!node_modules/**/*',
-    ];
-
-    logger.debug(`shell: ${shell}`);
-    logger.debug(`entrypoint: ${entrypoint}`);
-    logger.debug(`dependencies: ${dependencies}`);
+    let polymerProject = new PolymerProject({
+      root: config.root,
```
Comment (justinfagnani) [inline src/build/build.ts]: config can be null

File: src/build/build.ts [hunk 3]
```diff
@@ -120,102 +82,95 @@ export function build(options?: BuildOptions, config?: ProjectConfig): Promise<a
...
 
-    let unbundledPhase = new ForkedVinylStream(allFiles)
+    let unbundledPhase = forkStream(buildStream)
       .once('data', () => { logger.info('Generating build/unbundled...'); })
       .pipe(
         gulpif(
           options.insertDependencyLinks,
-          new PrefetchTransform(root, entrypoint,
-                shell, fragments, analyzer)
+          new PrefetchTransform(polymerProject.root, polymerProject.entrypoint,
```
Comment (justinfagnani) [inline src/build/build.ts]: It might be nice if things like this just depended on PolymerProject, or a project config

**Comment by justinfagnani** [comment]
LGTM

File: src/build/build.ts (hunk 3 above)
Comment (FredKSchott) [inline src/build/build.ts]: I intentionally haven't touched this in this PR, but totally agree on this point.

File: src/build/build.ts (hunk 2 above)
Comment (FredKSchott) [inline src/build/build.ts]: 👍  good catch

File: src/build/build.ts (hunk 1 above)
Comment (FredKSchott) [inline src/build/build.ts]: done

**Pull request merged**